app/main.py

```diff
@@ -1,13 +1,27 @@
-def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+from typing import List, Dict, Any
 
 
-def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+def format_linter_error(error: Dict[str, Any]) -> Dict[str, Any]:
+    return {
+        "line": error["line_number"],
+        "column": error["column_number"],
+        "message": error["text"],
+        "name": error["code"],
+        "source": "flake8"
+    }
 
 
-def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+def format_single_linter_file(
+        file_path: str,
+        errors: List[Dict[str, Any]]) -> Dict[str, Any]:
+    return {
+        "errors": [format_linter_error(error) for error in errors],
+        "path": file_path,
+        "status": "failed" if errors else "passed"
+    }
+
+
+def format_linter_report(
+        linter_report: Dict[str, Any]) -> List[Dict[str, Any]]:
```

Comment:
Please format this by placing closing parentheses on a new line for better readability